### PR TITLE
feat(fabric-socketio-connector): sending transactions signed on the client-side

### PR DESCRIPTION
**Pull request by outSH**
- Add functions to support entire process of sending transaction to the ledger without sharing
  any private key with the connector.
- Add test cases to check the new functionalities.
- Adjust asset-trade sample app to use new logic that doesn't send the private key to the connector.
- Move fabric signing utils to a separate file.
- Add fabric protobuf serializers and deserializers, so they can be send back and forth
  between connector and BLP without loosing type information.
- Add an option to FabricTestLedgerV1 to allow attaching to already running ledger container.
  This can speed up test development/troubleshoot process.
- Improve fabric-all-in-one ledger healthcheck, so that it waits for basic chaincode intialization.
  This is not strictly needed by current tests, but may be expected by future tests.
- Refactor duplicated logic in fabric-connector.
- Fix fabric-socketio module entry points (minor fix).

Closes: #2070

Depends on #2047

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

-------------

As always, please review only the last commit. Remaining ones are just dependencies, that will be squashed before an actual merge.

### CODE REVIEW

**Review by izuru0 — APPROVED** [review]
LGTM

File: examples/cactus-example-discounted-asset-trade/sign-utils.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
```
Comment (jagpreetsinghsasan) [inline examples/cactus-example-discounted-asset-trade/sign-utils.ts:1]: Wow, does HL fabric have documentation around this signing process? If yes, can you please attach some relevant fabric docs links about this, I would love to check them out.

File: examples/cactus-example-discounted-asset-trade/transaction-fabric.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      return resolve({
+        txId,
+        signedTxArgs
+      });
+    } catch (e) {
+      logger.error(`error at Invoke: err=${e}`);
+      return reject(e);
+    }
+  });
+}
```
Comment (jagpreetsinghsasan) [inline examples/cactus-example-discounted-asset-trade/transaction-fabric.ts:148]: Cool.

File: packages/cactus-plugin-ledger-connector-fabric-socketio/src/main/typescript/connector/fabric-proto-serializers.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    let proposalResponseMessage = new ProposalResponseType(proposalResponseCopy);
+    const encodedProposalResponse = proposalResponseMessage.encodeJSON();
+    return encodeMetdata(proposalResponseTypeName, encodedProposalResponse);
+  }
+
+  export function decode(encodedProposalResponse: string) {
+    let decodedProposalResponse = ProposalResponseType.decodeJSON(decodeMetdata(proposalResponseTypeName, encodedProposalResponse));
+    return convertToPlainObject(decodedProposalResponse);
+  }
+}
```
Comment (jagpreetsinghsasan) [inline packages/cactus-plugin-ledger-connector-fabric-socketio/src/main/typescript/connector/fabric-proto-serializers.ts:146]: Coool

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM.
And nice learning from this.

File: examples/cactus-example-discounted-asset-trade/sign-utils.ts (hunk 1 above)
Comment (outSH) [reply examples/cactus-example-discounted-asset-trade/sign-utils.ts:1]: @jagpreetsinghsasan Sure, the process is described here very well - https://hyperledger.github.io/fabric-sdk-node/release-1.4/tutorial-sign-transaction-offline.html

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH Please squash + rebase onto upstream main (and resolve the merge conflicts)

**Comment by outSH** [comment]
@petermetz Rebased and tested, ready for merge.

**Review by petermetz — APPROVED** [review]
@outSH LGTM, thank you!

**Pull request merged**